demos/mqtt/iot_demo_mqtt.c

```diff
@@ -719,30 +719,6 @@ static int _publishAllMessages( IotMqttConnection_t mqttConnection,
         }
     }
 
-    /* Wait for the messages in the last burst to be received. This should also
-     * wait for all previously published messages. */
-    if( status == EXIT_SUCCESS )
-    {
-        IotLogInfo( "Waiting for all publishes to be received." );
-
-        for( i = 0; i < IOT_DEMO_MQTT_PUBLISH_BURST_SIZE; i++ )
-        {
-            if( IotSemaphore_TimedWait( pPublishReceivedCounter,
-                                        MQTT_TIMEOUT_MS ) == false )
-            {
-                IotLogError( "Timed out waiting for incoming PUBLISH messages." );
-                status = EXIT_FAILURE;
-
-                break;
-            }
-        }
-
-        if( i == IOT_DEMO_MQTT_PUBLISH_BURST_SIZE )
-        {
-            IotLogInfo( "All publishes received." );
-        }
-    }
-
```

Comment:
Is this not needed?

Reply:
Correct, the change to the calculation above means this block is no longer needed.